src/merge_bed.py

```diff
@@ -27,12 +27,28 @@ def read_bed_file(f: TextIO) -> list[BedLine]:
         res.append(feature)
 
     return res
+    
+def bed_before(elem1: BedLine, elem2: BedLine) -> bool:
+    for bedline in ['chrom', 'chrom_start', 'chrom_end']:
+        if getattr(elem1,bedline) > getattr(elem2,bedline):
+            return False
+    return True
 
+def merge_lists_generateor(list1: list[BedLine], list2: [BedLine]) -> Generator[BedLine, None, None]:
+    iter1,iter2= iter(list1), iter(list2)
+    for (elem1, elem2) in zip(iter1,iter2):
+        if bed_before(elem1,elem2):
+            yield elem1
+        yield elem2
+    for elem1 in iter1:
+        yield elem1
+    for elem2 in iter2:
+        yield elem2
```

Comment:
If this works, it is by pure luck. Unfortunately, 'cause I like the thinking behind it.

The issue is that you take two sequences and zip them, so you get pairs of matched elements, and then you order the pairs. But that won't work if, for example, several elements from one sequence are smaller than the next in the other.

Consider sequences

```
x = [1, 2, 3, 4]
y = [7, 8, 9]
```

If you zip them, you get the pairs

```
(1, 7)
(2, 8)
(3, 9)
```

where `4` isn't included by `zip` by default.

If you order these pair-wise you get

```
1, 7, 2, 8, 3, 9
```

which is not what you want.